evm/utils/secp256k1.py

```diff
@@ -17,15 +17,16 @@
     pad32,
 )
 
+from evm.validation import (
+    validate_raw_public_key
+)
+
 
 def decode_public_key(public_key):
-    if len(public_key) != 64:
-        raise ValueError(
-            "Unexpected public key format: {}. Public keys must be 64 bytes long and must not "
-            "include the fixed \x04 prefix".format(public_key))
-    left = big_endian_to_int(public_key[0:32])
-    right = big_endian_to_int(public_key[32:64])
-    return left, right
+    if validate_raw_public_key(public_key):
```

Comment:
This should not be nested in an `if` statement.  Instead it should just be called at the beginning of the function.  Same with the other applications of this validator.